core/session.py

```diff
@@ -177,32 +175,29 @@ def __setitem__(self, key: str, item: Any):
     def markChanged(self) -> None:
         """
             Explicitly mark the current session as changed.
-            This will force save() to write into the memcache /
-            datastore, even if it belives that this session had
-            not changed.
+            This will force save() to write into the memcache / datastore,
+            even if it believes that this session hasn't changed.
         """
         self.changed = True
 
     def reset(self) -> None:
         """
-            Invalids the current session and starts a new one.
+            Invalidates the current session and starts a new one.
 
             This function is especially useful at login, where
             we might need to create an SSL-capable session.
 
             :warning: Everything (except the current language) is flushed.
```

Comment:
```suggestion
            :warning: Everything is flushed.
```
would be now correct for this method.

But I'm unsure about the removal. I rather think we should keep it here. Because the persistent fields are only considered on login/out in the user module. If a session is identified as invalid (because it's too old) and we reset it, we should keep the old language.

Reply:
Maybe we should add another conf-variable for this?

Reply:
Hmm. I don't know if we should always introduce config variables for everything...
But instead of a list of fields, maybe one should be able to inherit from the session class, adjust the reset method and then register the new class — if you need it.

Maybe we should test this on multilingual projects. For me we can leave it like this for now.

Reply:
> Maybe we should test this on multilingual projects. For me we can leave it like this for now.

Ok, then this is gonna be resolved.